### ISSUE TITLE
wsrep_debug=on discloses passwords in clear text

### ISSUE THREAD

**Issue by cr-bruderer** [mariadb-corporation/mysql-wsrep#216]
If wsrep_debug is set to on, passwords will be disclosed to the error log in clear text.

I consider this as a security threat.

**Comment by philip-galera** [comment]
Can you please paste a fragment of the log where the password is visible?

**Comment by cr-bruderer** [comment]
Fragment with the password visible:

2015-10-23 13:52:13 7434 [Warning] WSREP: last inactive check more than PT1.5S ago (PT2.49776S), skipping check
2015-10-23 13:54:06 7434 [Note] WSREP: TO BEGIN: -1, 0 : create user 'user'@'%' identified by 'password'
2015-10-23 13:54:06 7434 [Note] WSREP: TO BEGIN: 5, 2
2015-10-23 13:54:06 7434 [Note] WSREP: TO END: 5, 2 : create user 'user'@'%' identified by 'password'
2015-10-23 13:54:06 7434 [Note] WSREP: Set WSREPXid for InnoDB:  798b3473-7737-11e5-93f9-83d09a44e01b:5
2015-10-23 13:54:06 7434 [Note] WSREP: TO END: 5, update seqno
2015-10-23 13:54:06 7434 [Note] WSREP: TO END: 5

**Comment by philip-galera** [comment]
Turns out mysqld already provides a query rewriting facility that is used when writing the query to the general log. So a code like his would do the trick, though there are numerous log outputs that contain f thd->query():

```
diff --git a/sql/wsrep_mysqld.cc b/sql/wsrep_mysqld.cc
index 5dc8f93..263017f 100644
--- a/sql/wsrep_mysqld.cc
+++ b/sql/wsrep_mysqld.cc
@@ -1185,7 +1185,7 @@ static int wsrep_TOI_begin(THD *thd, char *db_, char *table_,
   int buf_err;

   WSREP_DEBUG("TO BEGIN: %lld, %d : %s", (long long)wsrep_thd_trx_seqno(thd),
-              thd->wsrep_exec_mode, thd->query() );
+              thd->wsrep_exec_mode, (!opt_log_raw) && thd->rewritten_query.length() ? thd->rewritten_query.c_ptr_safe() : thd->query() );
   switch (thd->lex->sql_command)
   {
   case SQLCOM_CREATE_VIEW:
```

and the following output is produced:

```
2015-11-12 10:48:22 14105 [Note] WSREP: TO BEGIN: -1, 0 : CREATE USER 'user'@'%' IDENTIFIED BY PASSWORD '*2470C0C06DEE42FD1618BB99005ADCA2EC9D1E19'
```

**Comment by philip-galera** [comment]
(The --log-raw option prevents rewriting)

**Comment by philip-galera** [comment]
Fixed in 5.6.28-25.13
